flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/slotpool/SlotPoolRpcTest.java

```diff
@@ -1,428 +1,428 @@
+//		} finally {
+//			RpcUtils.terminateRpcEndpoint(pool, timeout);
+//		}
+//	}
+//
+//	/**
+//	 * Testing SlotPool which exposes internal state via some testing methods.
+//	 */
+//	private static final class TestingSlotPool extends SlotPool {
+//
+//		private volatile Consumer<SlotRequestId> releaseSlotConsumer;
+//
+//		private volatile Consumer<SlotRequestId> timeoutPendingSlotRequestConsumer;
+//
+//		public TestingSlotPool(
+//				RpcService rpcService,
+//				JobID jobId,
+//				Clock clock,
+//				Time rpcTimeout,
+//				Time idleSlotTimeout) {
+//			super(
+//				rpcService,
+//				jobId,
+//				LocationPreferenceSchedulingStrategy.getInstance(),
+//				clock,
+//				rpcTimeout,
+//				idleSlotTimeout);
+//
+//			releaseSlotConsumer = null;
+//			timeoutPendingSlotRequestConsumer = null;
+//		}
+//
+//		public void setReleaseSlotConsumer(Consumer<SlotRequestId> releaseSlotConsumer) {
+//			this.releaseSlotConsumer = Preconditions.checkNotNull(releaseSlotConsumer);
+//		}
+//
+//		public void setTimeoutPendingSlotRequestConsumer(Consumer<SlotRequestId> timeoutPendingSlotRequestConsumer) {
+//			this.timeoutPendingSlotRequestConsumer = Preconditions.checkNotNull(timeoutPendingSlotRequestConsumer);
+//		}
+//
+//		@Override
+//		public CompletableFuture<Acknowledge> releaseSlot(
+//			SlotRequestId slotRequestId,
+//			@Nullable SlotSharingGroupId slotSharingGroupId,
+//			@Nullable Throwable cause) {
+//			final Consumer<SlotRequestId> currentReleaseSlotConsumer = releaseSlotConsumer;
+//
+//			final CompletableFuture<Acknowledge> acknowledgeCompletableFuture = super.releaseSlot(slotRequestId, slotSharingGroupId, cause);
+//
+//			if (currentReleaseSlotConsumer != null) {
+//				currentReleaseSlotConsumer.accept(slotRequestId);
+//			}
+//
+//			return acknowledgeCompletableFuture;
+//		}
+//
+//		@Override
+//		protected void timeoutPendingSlotRequest(SlotRequestId slotRequestId) {
+//			final Consumer<SlotRequestId> currentTimeoutPendingSlotRequestConsumer = timeoutPendingSlotRequestConsumer;
+//
+//			if (currentTimeoutPendingSlotRequestConsumer != null) {
+//				currentTimeoutPendingSlotRequestConsumer.accept(slotRequestId);
+//			}
+//
+//			super.timeoutPendingSlotRequest(slotRequestId);
+//		}
+//
+//		CompletableFuture<Boolean> containsAllocatedSlot(AllocationID allocationId) {
+//			return callAsync(
+//				() -> getAllocatedSlots().contains(allocationId),
+//				timeout);
+//		}
+//
+//		CompletableFuture<Boolean> containsAvailableSlot(AllocationID allocationId) {
+//			return callAsync(
+//				() -> getAvailableSlots().contains(allocationId),
+//				timeout);
+//		}
+//
+//		CompletableFuture<Integer> getNumberOfPendingRequests() {
+//			return callAsync(
+//				() -> getPendingRequests().size(),
+//				timeout);
+//		}
+//
+//		CompletableFuture<Integer> getNumberOfWaitingForResourceRequests() {
+//			return callAsync(
+//				() -> getWaitingForResourceManager().size(),
+//				timeout);
+//		}
+//	}
+//
+//}
```

Comment:
I think we need to port these tests to the new architecture using the `Scheduler` or simply making them new tests in `SlotPoolTest`.